### ISSUE TITLE
missing stdint.h header in Fl_Cairo_Graphics_Driver.cxx when building with gcc 5

### ISSUE THREAD

**Issue by MoAlyousef** [fltk/fltk#522]
**Describe the bug**
Encountered this when building with gcc-5 (because of reasons beyond my control!). 

```
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx: In member function 'virtual void Fl_Cairo_Graphics_Driver::cache(Fl_RGB_Image*)':
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx:838:11: error: 'uint32_t' was not declared in this scope
           *(uint32_t *)q = A << 24 | (uchar)(R*f) << 16 | (uchar)(G*f) << 8 | (uchar)(B*f);
             ^
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx:838:21: error: expected primary-expression before ')' token
           *(uint32_t *)q = A << 24 | (uchar)(R*f) << 16 | (uchar)(G*f) << 8 | (uchar)(B*f);
                       ^
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx:854:11: error: 'uint32_t' was not declared in this scope
           *(uint32_t *)q = A << 24 | G << 16 | G << 8 | G;
             ^
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx:854:21: error: expected primary-expression before ')' token
           *(uint32_t *)q = A << 24 | G << 16 | G << 8 | G;
                       ^
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx: In static member function 'static cairo_pattern_t* Fl_Cairo_Graphics_Driver::bitmap_to_pattern(Fl_Bitmap*, bool, cairo_surface_t**)':
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx:903:5: error: 'uint32_t' was not declared in this scope
       uint32_t *q = (uint32_t *)(BGRA + j * stride);
       ^
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx:903:15: error: 'q' was not declared in this scope
       uint32_t *q = (uint32_t *)(BGRA + j * stride);
                 ^
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx:903:30: error: expected primary-expression before ')' token
       uint32_t *q = (uint32_t *)(BGRA + j * stride);
                                ^
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx:905:14: error: expected ';' before 'mask32'
       uint32_t mask32;
                ^
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx:912:9: error: 'mask32' was not declared in this scope
           mask32 = 1;
           ^
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx:921:17: error: 'mask32' was not declared in this scope
           (*q) |= mask32;
                   ^
  /home/rayloom/Documents/dev/app/fltk/src/drivers/Cairo/Fl_Cairo_Graphics_Driver.cxx:927:7: error: 'mask32' was not declared in this scope
         mask32 <<= 1;
```

**To Reproduce**
cmake -Bbin -DOPTION_USE_CAIRO=ON -DCMAKE_BUILD_TYPE=Release && cmake --build bin --parallel

N/A

**Expected behavior**
The build should succeed.

**Screenshots**
N/A

**FLTK Version**
 - Version:1.4.0
 - Git, branch: master

**FLTK Configure / Build Options**
cmake -Bbin -DOPTION_USE_CAIRO=ON -DCMAKE_BUILD_TYPE=Release && cmake --build bin --parallel

**Operating System / Platform:**
Linux: Ubuntu 20.04

**Linux/Unix Runtime, if applicable:**
 - X11

Adding the stdint.h header fixes it.

**Comment by erco77** [comment]
I'm guessing the problem is in `FL/platform_types.h`; there appears to be some `#ifdef` guards preventing the include for `stdint.h` to occur, which defines the `uint32_t` type that is causing the error.

OP, to just simply "get by", try adding `#include <stdint.h>` to the top of `FL/platform_types.h` and see if you get through the build OK.

It seems the intended path through the includes should be:
```
Fl_Cairo_Graphics_Driver.cxx: #include <FL/platform.H>
FL/platform.H: #include <FL/platform_types.h>
FL/platform_types.h: # include<stdint.h>
```
There are some `#ifdef` guards that might be preventing this, so I figure Albrecht or Manolo might want to tweak the guards.

**Comment by MoAlyousef** [comment]
Including stdint.h into FL/platform_types.h also works.

**Comment by ManoloFLTK** [comment]
@MoAlyousef That should be fixed with commit b0d4c2f . Please, close the issue if you confirm.

**Comment by MoAlyousef** [comment]
Yes. Thank you
